### PR TITLE
Uncolorize winston when stdout isn't a tty

### LINKED ISSUE #5139: [REQUEST]: dont colorize log message when not started in a console
### Is your feature request related to a problem? Please describe

I launch compiler-explorer from a GUI, that GUI then read the output, and it seems that the escapes added to colourize the log messages are still emitted.

example:

log_colors

Usually developmenet tools have a system allowing to deactivate colouring when not launched from a terminal.

### Describe the solution you'd like

There should be a system to deactivate coloring, either a manual one using extra args or an automatic one based on `isatty()`. 

Other option is to support [No_COLOR](https://no-color.org/)

### Describe alternatives you've considered

none, this is a minor problem

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Closes #5139

### CODE REVIEW

File: lib/logger.ts [hunk 1]
```diff
@@ -33,7 +33,11 @@ import TransportStream, {TransportStreamOptions} from 'winston-transport';
 
 const consoleTransportInstance = new winston.transports.Console();
 export const logger = winston.createLogger({
-    format: winston.format.combine(winston.format.colorize(), winston.format.splat(), winston.format.simple()),
+    format: winston.format.combine(
+        process.stdout.isTTY ? winston.format.colorize() : winston.format.uncolorize(),
+        winston.format.splat(),
+        winston.format.simple(),
+    ),
```
Comment (jeremy-rifkin) [inline lib/logger.ts:40]: Will this cause any problems on papertrail or whatnot? cc @mattgodbolt

**Comment by jeremy-rifkin** [comment]
This does appear to interfere with papertrail / how things are setup on aws. I'm not sure the best solution.

**Comment by partouf** [comment]
> This does appear to interfere with papertrail / how things are setup on aws. I'm not sure the best solution.

I'm guessing that you would need to recreate `logger` maybe when `logToPapertrail` is called?
Or perhaps you can get into the internals and change the formatting back to normal?

**Comment by partouf** [comment]
> > This does appear to interfere with papertrail / how things are setup on aws. I'm not sure the best solution.
> 
> I'm guessing that you would need to recreate `logger` maybe when `logToPapertrail` is called? Or perhaps you can get into the internals and change the formatting back to normal?

OH

Or maybe create a new function to solidify writing to console and do the same that we do with the Loki logger?
https://github.com/compiler-explorer/compiler-explorer/blob/c04512109ccce9de4da2c1558147f548b010163c/lib/logger.ts#L107

**Comment by partouf** [comment]
You can remove existing transports by doing `.remove()` https://github.com/winstonjs/winston#further-reading

**Comment by jeremy-rifkin** [comment]
Ope I did it wrong

**Comment by jeremy-rifkin** [comment]
I think this'll work now. Tested in staging and it looks good.

**Comment by partouf** [comment]
now live

**Pull request merged**